sklearn/datasets/openml.py

```diff
@@ -0,0 +1,495 @@
+        json_data = _get_json_content_from_openml_api(url, error_msg, True)
+
+    return json_data['data']['dataset'][0]
+
+
+def _get_data_description_by_id(data_id):
+    # OpenML API function: https://www.openml.org/api_docs#!/data/get_data_id
+    url = _DATA_INFO.format(data_id)
+    error_message = "Dataset with data_id {} not found.".format(data_id)
+    json_data = _get_json_content_from_openml_api(url, error_message, True)
+    return json_data['data_set_description']
+
+
+def _get_data_features(data_id):
+    # OpenML function:
+    # https://www.openml.org/api_docs#!/data/get_data_features_id
+    url = _DATA_FEATURES.format(data_id)
+    error_message = "Dataset with data_id {} not found.".format(data_id)
+    json_data = _get_json_content_from_openml_api(url, error_message, True)
+    return json_data['data_features']['feature']
+
+
+def _download_data_arff(file_id, sparse):
+    # Accesses an ARFF file on the OpenML server. Documentation:
+    # https://www.openml.org/api_data_docs#!/data/get_download_id
+    url = _DATA_FILE.format(file_id)
+    response = urlopen(url)
+    kwargs = {}
+    if sparse is True:
+        kwargs['return_type'] = arff.COO
+    else:
+        kwargs['return_type'] = arff.DENSE
+
+    if PY2:
+        # Because Python2.7 numpy can't handle unicode
+        arff_file = arff.loads(response.read(), **kwargs)
+    else:
+        arff_file = arff.loads(response.read().decode('utf-8'), **kwargs)
+
+    response.close()
+    return arff_file
+
+
+def _determine_default_target(features_list):
+    # determines the default target based on the data feature results
+    # (which is currently more reliable than the data description;
+    # see issue: https://github.com/openml/OpenML/issues/768)
+    results = [feature['name']
+               for feature in features_list if feature['is_target'] == 'true']
+
+    if len(results) == 0:
+        return None
+    elif len(results) == 1:
+        return results[0]
+    else:
+        return results
+
+
+def _determine_single_target_data_type(features_dict, target_column_name):
+    # determine the data type of the y array in case there is a single target
+    if not isinstance(target_column_name, string_types):
+        raise ValueError('target_column_name should be of string type, '
+                         'got: %s' % type(target_column_name))
+    if target_column_name not in features_dict:
+        raise KeyError('Could not find target_column_name={}')
+    feature = features_dict[target_column_name]
+    # note: we compare to a string, not boolean
+
+    for dict_key in ['is_ignore', 'is_row_identifier']:
+        if feature[dict_key] == 'true':
+            warn('target_column_name={} has flag {}.'.format(
+                target_column_name, dict_key))
+
+    if feature['data_type'] == "numeric":
+        return np.float64
+    else:
+        return object
+
+
+def _determine_multi_target_data_type(features_dict, target_column_names):
```

Comment:
I think this now can be renamed

Reply:
Agreed. To _determine_target_data_type() ?